marc.go

```diff
@@ -313,3 +317,49 @@ func contentType(x byte) string {
 	}
 	return t
 }
+
+// RetrieveLanguageCodelist retrieves language codes for parsing MARC languages
+func RetrieveLanguageCodelist() (map[string]string, error) {
+	file, err := os.Open("fixtures/languages.xml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Language struct
+	type Language struct {
+		Name string `xml:"name"`
+		Code string `xml:"code"`
+	}
+
+	decoder := xml.NewDecoder(file)
+	languages := make(map[string]string)
+
+	for {
+		t, _ := decoder.Token()
+		if t == nil {
+			break
+		}
+		switch se := t.(type) {
+		case xml.StartElement:
+			if se.Name.Local == "language" {
+				var l Language
+				decoder.DecodeElement(&l, &se)
+				languages[l.Code] = l.Name
+			}
+		}
+	}
+	return languages, err
+}
+
+// TranslateLanguageCodes takes an array of MARC language codes and returns the language names.
+func TranslateLanguageCodes(x []string, languageCodes map[string]string) []string {
+	var languages []string
```

Comment:
Maybe just call `RetrieveLanguageCodelist` here instead of including in the parser since it's only needed for this?

Reply:
I was trying to have it only process and store the XML file once per parser call, rather than for every record, for efficiency purposes. But I realize now that maybe that's not how channels work in Go? If it's called in the Generator channel it gets called for every record anyway, is that correct?

Reply:
That's a good question. I'm not opposed to leaving it how you have it if that is the case. The method calls just got a bit uglier and I wanted them not to... particularly as you'll be adding another lookup and it's going to get messy.